v1/event_format/formatter.h

```diff
@@ -14,17 +14,17 @@ namespace format {
  * StructuredCloudEvents (serializations based on EventFormats)
  */
 class Formatter {
-    public:
-        // Marshal a CloudEvent into a StructuredCloudEvent
-        virtual absl::StatusOr<StructuredCloudEvent> Serialize(
-            const io::cloudevents::v1::CloudEvent& cloud_event) = 0;
-        
-        // Marshal a StructuredCloudEvent into a CloudEvent
-        virtual absl::StatusOr<io::cloudevents::v1::CloudEvent> Deserialize(
-            const StructuredCloudEvent& structured_cloud_event) = 0;
+ public:
+  // Marshal a CloudEvent into a StructuredCloudEvent
+  virtual absl::StatusOr<std::unique_ptr<StructuredCloudEvent>> Serialize(
```

Comment:
https://blog.ycshao.com/2012/03/18/effective-c-item-7-declare-destructors-virtual-in-polymorphic-base-classes/
'Polymorphic base classes should declare virtual destructors. If a class has any virtual functions, it should have a virtual destructor.'